contracts/interfaces/IPeriphery.sol

```diff
@@ -8,11 +8,71 @@ import "../utils/CustomErrors.sol";
 import "../interfaces/IWETH.sol";
 
 interface IPeriphery is CustomErrors {
-    // events
+    // structs for events
+    struct PositionDetails {
+        IMarginEngine marginEngine;
+        address owner;
+        int24 tickLower;
+        int24 tickUpper;
+    }
+
+    struct SwapOutput {
+        int256 variableTokenDelta;
+        uint256 cumulativeFeeIncurred;
+        int256 marginRequirement;
```

Comment:
is there a way to track both the initial and liquidation margin requirements here?

Reply:
unfortunately, that's not possible. we don't have access to liquidation threshold.